state/machine_linklayerdevices.go

```diff
@@ -187,13 +187,13 @@ func (m *Machine) SetLinkLayerDevices(devicesArgs ...LinkLayerDeviceArgs) (err e
 			return nil, errors.Trace(err)
 		}
 
+		if err := checkModelActive(m.st); err != nil {
+			return nil, errors.Trace(err)
+		}
+		if err := m.isStillAlive(); err != nil {
```

Comment:
it would be a little more efficient to use the m.doc.Life value here, and only reload the machine doc on attempt>0. I suspect that's what was intended originally, except the initial check was left out